tests/iam/certhandler/certhandler_test.cpp

```diff
@@ -6,15 +6,523 @@
  */
 
 #include <gtest/gtest.h>
+#include <iostream>
+#include <vector>
 
 #include "aos/iam/certhandler.hpp"
 
+#include "mocks/hsm.hpp"
+#include "mocks/storage.hpp"
+#include "mocks/x509provider.hpp"
+
 using aos::Error;
-using aos::iam::certhandler::CertHandler;
+using namespace aos;
+using namespace aos::iam::certhandler;
+using namespace ::testing;
+
+class CertHandlerTest : public ::testing::Test {
+protected:
+    virtual void SetUp()
+    {
+        EXPECT_CALL(mX509Provider, DNToString(_, _))
+            .WillRepeatedly(Invoke([](const Array<uint8_t>& dn, String& result) {
+                result = reinterpret_cast<const char*>(dn.begin());
+                return Error::Enum::eNone;
+            }));
+    }
+
+    void ApplyCert(const Array<crypto::x509::Certificate>& certChain, const CertInfo& devCertInfo)
+    {
+        EXPECT_CALL(mX509Provider, PEMToX509Certs(_, _))
+            .WillOnce(DoAll(SetArgReferee<1>(certChain), Return(Error::Enum::eNone)));
+        EXPECT_CALL(mPKCS11, ApplyCert(_, _, _))
+            .WillOnce(DoAll(SetArgReferee<1>(devCertInfo), SetArgReferee<2>(cPassword), Return(Error::Enum::eNone)));
+        EXPECT_CALL(mStorage, AddCertInfo(cPKCS11Type, devCertInfo)).WillOnce(Return(Error::Enum::eNone));
+
+        StaticArray<uint8_t, 1> pemCert;
+        CertInfo                certInfo;
+        EXPECT_EQ(Error::Enum::eNone, mCertHandler.ApplyCertificate(cPKCS11Type, pemCert, certInfo));
+    }
+
+    // certmodule class dependendcies
+    MockHSMItf                    mPKCS11, mSW, mTPM;
+    crypto::x509::MockProviderItf mX509Provider;
+    // cerhandler class dependencies
+    MockStorageItf mStorage;
+
+    CertHandler mCertHandler;
+
+    static const ModuleConfig cDefaultConfig;
+
+    static const String cPKCS11Type;
+    static const String cSWType;
+    static const String cTPMType;
+
+    static const String cPassword;
+
+    static const Array<uint8_t> cIssuer;
+    static const Array<uint8_t> cSubject;
+    static crypto::PrivateKey   cPrivateKey;
```

Comment:
Move const to top of protected sections: first const, methods, vars

Reply:
done